allennlp/models/semantic_parsing/drop/naqanet.py

```diff
@@ -78,11 +93,31 @@ def __init__(self, vocab: Vocabulary,
         self._drop_metrics = DropEmAndF1()
         self._dropout = torch.nn.Dropout(p=dropout_prob)
 
+        self._rule_namespace = rule_namespace
+        self._add_action_bias = add_action_bias
+
         num_actions = vocab.get_vocab_size(self._rule_namespace)
+
+        self.naqanet_parameters = NaqanetParameters(modeling_dim)
+        self._first_action_embedding = torch.nn.Parameter(torch.FloatTensor(action_embedding_dim))
+
+
         if self._add_action_bias:
             self._action_biases = Embedding(num_embeddings=num_actions, embedding_dim=1)
         self._action_embedder = Embedding(num_embeddings=num_actions, embedding_dim=action_embedding_dim)
 
+        self._decoder_beam_search = decoder_beam_search
+        self._max_decoding_steps = max_decoding_steps
+        
+        self._transition_function = LinkingTransitionFunction(encoder_output_dim=encoding_out_dim,
+                                                              action_embedding_dim=action_embedding_dim,
+                                                              input_attention=input_attention,
+                                                              predict_start_type_separately=False,
```

Comment:
I think you should set this to `True`, and remove the START_SYMBOL.

Reply:
I tried doing this but executing requires the START SYMBOL (https://github.com/allenai/allennlp/blob/master/allennlp/semparse/domain_languages/domain_language.py#L329). Should we remove this check in domain languages, or postprocess our actions here to include `[START_SYMBOL -> Answer]`?

Reply:
Yeah, one option is to just prepend that action before calling `execute_action_sequence`.